tests/lib/config.js

```diff
@@ -884,6 +884,38 @@ describe("Config", function() {
             assertConfigsEqual(actual, expected);
         });
 
+        it("should extend package sub-configuration with prefix", function() {
+
+            var configDeps = {};
+            configDeps["./config/config-file"] = proxyquire("../../lib/config/config-file", {
+                "@scope/eslint-config/foo": {
+                    rules: {
+                        eqeqeq: 2
+                    }
+                }
+            });
+
+            var StubbedConfig = proxyquire("../../lib/config", configDeps);
+
+            var configPath = path.resolve(__dirname, "../fixtures/config-extends/scoped-package9/.eslintrc"),
+                configHelper = new StubbedConfig({ useEslintrc: true, configFile: configPath, cwd: process.cwd() }),
+                expected = {
+                    rules: { "quotes": [2, "double"], "eqeqeq": 2, "valid-jsdoc": 0 },
+                    env: { "browser": false }
+                };
+
+            // Reason to override this function in this special case is that I don't want it to keep looking into the
+            // chain of parent directories for .eslintrc or package.json file for configs.
+            // If that is allowed then the expected outcome will change based on the machine you are running this test.
+            sinon.stub(configHelper, "findLocalConfigFiles").returns([
+                path.resolve(__dirname, "../fixtures/config-extends/scoped-package9/.eslintrc")
```

Comment:
Long time down the memory lane, to provide the config file as well have `.eslintrc` file is perfectly fine. As in that case it will again reload the same config path because of `.eslintrc` set to true. The line which said that its a noop is not noop as the unit test is trying to test the caching issue we had before. Thats why I call the `getConfig` function on the same object of class more than once.